sys/shell/commands/sc_gnrc_icmpv6_echo.c

```diff
@@ -171,17 +171,16 @@ static int _configure(int argc, char **argv, _ping_data_t *data)
     for (int i = 1; i < argc; i++) {
         char *arg = argv[i];
         if (arg[0] != '-') {
-            int iface;
 
             data->hostname = arg;
 #ifdef MODULE_SOCK_DNS
             if (sock_dns_query(data->hostname, &data->host, AF_INET6) == 0) {
                 continue;
             }
 #endif
-            iface = ipv6_addr_split_iface(data->hostname);
-            if (iface != -1) {
-                data->netif = gnrc_netif_get_by_pid(iface);
+            char *iface = ipv6_addr_split_iface(data->hostname);
+            if (iface) {
```

Comment:
Is this check working? From the code I would expect `""` is returned when no interface is given. This check implies a `NULL` pointer.

Reply:
True, I will modify `ipv6_addr_split_str` to return `NULL` when the separator was not found

Reply:
That way it's more similar to `strchr` also I guess

Reply:
I would more compare it to `strtok()`, but the return value is the same ;-)